### PR TITLE
Add 3D version of UpSampling

### PR DESCRIPTION
**Pull request by tanmayb123**

### CODE REVIEW

File: Sources/DeepLearning/Layer.swift [hunk 1]
```diff
@@ -1069,6 +1069,35 @@ public struct UpSampling2D<Scalar: TensorFlowFloatingPoint>: Layer {
...
+    ///
+    /// - Parameters:
+    ///   - input: The input to the layer.
+    ///   - context: The contextual information for the layer application, e.g. the current learning
+    ///     phase.
+    /// - Returns: The output.
+    @differentiable
+    public func applied(to input: Tensor<Scalar>, in _: Context) -> Tensor<Scalar> {
+        let shape = input.shape
+        let (batchSize, height, width, depth, channels) = (shape[0], shape[1], shape[2], shape[3])
```
Comment (rxwei) [inline Sources/DeepLearning/Layer.swift]: This line is not compiling.

Comment (tanmayb123) [reply Sources/DeepLearning/Layer.swift]: Ah I had committed an earlier version. Sorry!

Comment (tanmayb123) [reply Sources/DeepLearning/Layer.swift]: Fixed.

File: Sources/DeepLearning/Layer.swift [hunk 2]
```diff
@@ -1069,6 +1069,38 @@ public struct UpSampling2D<Scalar: TensorFlowFloatingPoint>: Layer {
...
+    /// - Returns: The output.
+    @differentiable
+    public func applied(to input: Tensor<Scalar>, in _: Context) -> Tensor<Scalar> {
+        let shape = input.shape
+        let (batchSize, height, width, depth, channels) = (shape[0], shape[1], shape[2], shape[3], shape[4])
+        let scaleOnes2D = Tensor<Scalar>(ones: [1, 1, size, 1, size, 1, 1])
+        var upSampling2D = input.reshaped(to: [batchSize, height, 1, width, 1, depth, channels]) * scaleOnes2D
+        upSampling2D = upSampling2D.reshaped(to: [batchSize, height * size, width * size, depth, channels])
+        let scaleOnes3D = Tensor<Scalar>(ones: [1, 1, 1, 1, size, 1])
+        let upSampling3D = upSampling2D.reshaped(to: [batchSize, height * size, width * size, depth, 1, channels]) * scaleOnes3D
```
Comment (rxwei) [inline Sources/DeepLearning/Layer.swift]: Fit all lines within 100 columns.

Comment (tanmayb123) [reply Sources/DeepLearning/Layer.swift]: Sorry - fixed.

**Comment by tanmayb123** [comment]
Seems like the build failed because of PRNG stuff. Should I close this PR and create a new one?

**Comment by rxwei** [comment]
I believe all you need is to pull the latest master.

**Comment by tanmayb123** [comment]
I deleted this repository though. I'll just close/re-open.

**Pull request closed without merging**